common/djangoapps/xblock_django/user_service.py

```diff
@@ -22,6 +27,32 @@ def get_current_user(self):
         """
         return self._convert_django_user_to_xblock_user(self._django_user)
 
+    def get_anonymous_user_id(self, username, course_id):
+        """
+        Get the anonymous user id for a user.
+
+        Args:
+            username(str): username of a user.
+            course_id(str): course id of particular course.
+
+        Returns:
+            A unique anonymous_user_id for (user, course) pair.
+            None for Non-staff users.
+        """
+        if not self.get_current_user().opt_attrs.get(ATTR_KEY_USER_IS_STAFF):
+            return None
+
+        # If we import these in start, it causes the contentstore library tests failed.
+        from student.models import anonymous_id_for_user, get_user_by_username_or_email
```

Comment:
Having to hid an import inside a function like that is a bit of a red flag that perhaps the dependencies are pointing the wrong way.